Sources/Markdown/Interpretive Nodes/Aside.swift

```diff
@@ -170,51 +180,72 @@ public struct Aside {
             self.content = Array(blockQuote.blockChildren)
             return
         }
-        self.kind = Kind(rawValue: String(initialText.string[..<firstColonIndex]))!
 
-        // Trim off the aside tag prefix.
-        let trimmedText = initialText.string[initialText.string.index(after: firstColonIndex)...].drop {
-            $0 == " " || $0 == "\t"
+        self.kind = Kind(rawValue: kindTag.tag)!
+        self.content = Array(kindTag.newBlockQuote.blockChildren)
+    }
+
+    /// Create an aside from a block quote if it contains a kind tag that matches the given requirement.
+    public init?(_ blockQuote: BlockQuote, tagRequirement: TagRequirement = .tagNotRequired) {
+        guard tagRequirement != .tagNotRequired else {
+            self = .init(blockQuote)
+            return
+        }
+
+        guard let kindTag = blockQuote.parseAsideTag(tagRequirement: tagRequirement) else {
+            return nil
         }
-        initialText.string = String(trimmedText)
 
-        let newBlockQuote = initialText.parent!.parent! as! BlockQuote
-        self.content = Array(newBlockQuote.blockChildren)
+        self.kind = Kind(rawValue: kindTag.tag)!
+        self.content = Array(kindTag.newBlockQuote.blockChildren)
     }
 }
 
 extension BlockQuote {
-    /// Conservatively checks the text of this block quote to see whether it can be parsed as an aside.
-    ///
-    /// Whereas ``Aside/init(_:)`` will use all the text before the first colon in the first line,
-    /// or else return an ``Aside`` with a ``Aside/Kind`` of ``Aside/Kind/note``,
-    /// this function will allow parsers to only parse an aside if there is a single-word aside
-    /// marker in the first line, and otherwise fall back to a plain ``BlockQuote``.
-    func isAside() -> Bool {
+    func parseAsideTag(tagRequirement: Aside.TagRequirement) -> (tag: String, newBlockQuote: BlockQuote)? {
         guard let initialText = self.child(through: [
             (0, Paragraph.self),
             (0, Text.self),
-        ]) as? Text else {
-            return false
+        ]) as? Text, let firstColonIndex = initialText.string.firstIndex(of: ":") else {
+            return nil
         }
 
-        for character in initialText.string {
-            switch character {
-                case ":": return true  // encountered ":" before " "
-                case " ": return false // encountered " " before ":"
-                default:  continue
-            }
+        let kindTag = initialText.string[..<firstColonIndex]
+        let trimmedText = initialText.string[initialText.string.index(after: firstColonIndex)...].drop {
+            $0 == " " || $0 == "\t"
+        }
+
+        guard tagRequirement != .requireSingleWordTag || !kindTag.contains(" ") else {
+            return nil
         }
-        return false // didn't encounter either " " or ":" in the string
+
+        let shiftCount = kindTag.count + 1 + initialText.string[firstColonIndex...].dropFirst().prefix(while: {
```

Comment:
minor: `column` (which this adds to) is documented to be number of UTF-8 bytes. It's possible that a tag name in another language could cause an offset source range.
```suggestion
        let shiftCount = kindTag.utf8.count + 1 + initialText.string[firstColonIndex...].dropFirst().prefix(while: {
```